### PR TITLE
Guess the units of the vertical axis for 2D CRS

### PR DESCRIPTION
**Pull request by lilleyse**
For https://github.com/CesiumGS/asset-pipeline/pull/921

### CODE REVIEW

**Comment by lilleyse** [comment]
@ptrgags can you review?

File: sources/src/parser/geocoordinatetransformer.cpp [hunk 1]
```diff
@@ -100,6 +104,14 @@ class GeoTransform
         OGRErr err = sourceSRS.SetFromUserInput(sourceURN.c_str());
         sourceSRS.SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
 
+        // Guess the units of the vertical axis for 2D CRS
+        double verticalToMeters = 1.0;
+        int axesCount = sourceSRS.GetAxesCount();
+        if (axesCount == 2)
+        {
+            verticalToMeters = sourceSRS.GetLinearUnits();
```
Comment (ptrgags) [inline sources/src/parser/geocoordinatetransformer.cpp:112]: I don't know GDAL very well, what if the SRS is a cartographic SRS so the first two axes are angles? what will it return then?

**Review by ptrgags — COMMENTED** [review]
The code seems to make sense to me, though once I have the command to run unit tests I'll try that out.

I did have one question because I'm not familiar with GDAL

File: sources/src/parser/geocoordinatetransformer.cpp (hunk 1 above)
Comment (lilleyse) [reply sources/src/parser/geocoordinatetransformer.cpp:112]: There's separate getters for angular units and linear units. I believe linear units will default to meters if 2D axes are cartographic. I should confirm that this happens with point clouds.

Comment (ptrgags) [reply sources/src/parser/geocoordinatetransformer.cpp:112]: https://gdal.org/doxygen/classOGRSpatialReference.html#aefbcac403652e15cfcb1d864d84a3706 -- looks like it does default to 1.0 (meters).

Also it seems like this is deprecated in favor of a version that returns the units name

**Comment by ptrgags** [comment]
Ran the unit tests with asset-pipeline https://github.com/CesiumGS/asset-pipeline/pull/921 + this branch, all the tests passed

I'll test asset-pipeline master + this branch next to compare.

**Comment by ptrgags** [comment]
Tested that. Summary:

| asset-pipeline branch | libcitygml branch | results |
|------------------|--|--|
| update-gdal-proj  | fix-vertical-units | all tests pass |
| update-gdal-proj | master | NYC12.ReadProperties and NYC12.TotalPrimitivesAndTexels fail |
| master | fix-vertical-units | NYC12.ReadProperties and NYC12.TotalPrimitivesAndTexels fail |
| master | master | all tests pass |

So no surprises here. The two branches work properly together but we need to update both for asset-pipeline to work.

**Comment by ptrgags** [comment]
Looks good, thanks @lilleyse!

**Pull request merged**